ScrollOfResurrection.lua

```diff
@@ -0,0 +1,741 @@
+local BASE_DOW=4    -- 1970-01-01 was a Thursday
+local BASE_YEAR=1970    -- 1970 is the base year
+
+local _days={
+    -1, 30, 58, 89, 119, 150, 180, 211, 242, 272, 303, 333, 364
+}
+local _lpdays={}
+for i=1,2  do _lpdays[i]=_days[i]   end
+for i=3,13 do _lpdays[i]=_days[i]+1 end
+
+local function SOR_gmtime(t)
+    local y,j,m,d,w,h,n,s
+    local mdays=_days
+    s=t
+    -- First calculate the number of four-year-interval, so calculation
+    -- of leap year will be simple. Btw, because 2000 IS a leap year and
+    -- 2100 is out of range, this formula is so simple.
+    y=floor(s/FSEC)
+    s=s-y*FSEC
+    y=y*4+BASE_YEAR         -- 1970, 1974, 1978, ...
+    if s>=YSEC then
+        y=y+1           -- 1971, 1975, 1979,...
+        s=s-YSEC
+        if s>=YSEC then
+            y=y+1       -- 1972, 1976, 1980,... (leap years!)
+            s=s-YSEC
+            if s>=LSEC then
+                y=y+1   -- 1971, 1975, 1979,...
+                s=s-LSEC
+            else        -- leap year
+                mdays=_lpdays
+            end
+        end
+    end
+    j=floor(s/DSEC)
+    s=s-j*DSEC
+    local m=1
+    while mdays[m]<j do m=m+1 end
+    m=m-1
+    local d=j-mdays[m]
+    -- Calculate day of week. Sunday is 0
+    w=(floor(t/DSEC)+BASE_DOW)%7
+    -- Calculate the time of day from the remaining seconds
+    h=floor(s/3600)
+    s=s-h*3600
+    n=floor(s/60)
+    s=s-n*60
+    d = SOR_numeralise(d)
+    return(({"January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"})[m]..', '..d..' '..y..' '..h..':'..n..':'..s)
+end
+
+local function SOR_PreventReturn(playerGUID)
+    if Config.preventReturn == 1 then
+        CharDBExecute('UPDATE `mail` SET `messageType` = 3 WHERE `sender` = '..Config.senderGUID..' AND `receiver` = '..playerGUID..' AND `messageType` = 0;')
+    end
+end
+
+local function SOR_cleanup()
+    --todo: check variables for required cleanups
+    --set all non local runtime variables to nil
+end
+
+local function SOR_splitString(inputstr, seperator)
+    if seperator == nil then
+        seperator = "%s"
+    end
+    local t={}
+    for str in string.gmatch(inputstr, "([^"..seperator.."]+)") do
+        table.insert(t, str)
+    end
+    return t
+end
+
+local function SOR_hasValue (tab, val)
+    for index, value in ipairs(tab) do
+        if value == val then
+            return true
+        end
+    end
+    return false
+end
+
+local function SOR_hasIndex (tab, val)
+    for index, value in ipairs(tab) do
+        if index == val then
+            return true
+        end
+    end
+    return false
+end
+
+local function SOR_checkAbuse(accountId)
+    if SOR_abuseCounter[accountId] == nil then
+        SOR_abuseCounter[accountId] = 1
+    else
+        SOR_abuseCounter[accountId] = SOR_abuseCounter[accountId] + 1
+    end
+
+    if SOR_abuseCounter[accountId] > Config.abuseTreshold then
+        return true
+    end
+    return false
```

Comment:
can't return directly `SOR_abuseCounter[accountId] > Config.abuseTreshold`? it's a boolean condition, so it will return true/false based on its results.
`return SOR_abuseCounter[accountId] > Config.abuseTreshold`